### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by demiantres**
Improvements:
- const methods where appropriate
- Added QtFirebaseAdMobBase and QtFirebaseAdMobBannerBase classes

### CODE REVIEW

**Comment by Larpon** [comment]
@demiantres - thanks! A good idea with the base classes. I've skimmed your changes and they look fine. I'll have to check if everything is working in the example app (both Android + iOS) - and then merge if everything runs :)

**Comment by demiantres** [comment]
I have tested all the Admob ad types in iOS and here they work fine (but four eyes see more than two ;)).

File: src/qtfirebase.cpp [hunk 1]
```diff
@@ -60,6 +63,7 @@ void QtFirebase::waitForFutureCompletion(firebase::FutureBase future)
             count = 0;
             break;
         }
+        QThread::msleep(10);
```
Comment (Larpon) [inline src/qtfirebase.cpp]: I'm a bit unsure why you've put this here? :)

Comment (demiantres) [reply src/qtfirebase.cpp]: It lowers the CPU usage and gives a better way to cancel the loop (more independent of CPU speed). Also (I forgot to change this): the 100000 and 20000 numbers must be reduced to something like 100 (1 second) and 200 (2 seconds).

**Pull request merged**